coalib/bearlib/abstractions/Linter.py

```diff
@@ -122,6 +122,15 @@ def __repr__(cls):
                 cls.__name__, options["executable"])
 
     class LinterBase(LocalBear, metaclass=LinterMeta):
+        def __init__(self):
+            if options["output_format"] == "regex":
+                options["output_regex"] = re.compile(options["output_regex"])
+                if options["output_regex"].groups > 8:
+                    self.warn("Performance: Using unnecessary capture groups"
+                              " affects the performance of coala. "
+                              "You should use ``(?:<pattern>)`` instead of"
+                              " ``(<pattern>)`` for your regex")
+
```

Comment:
> >>> x = re.compile(".(.).")
>>> x.groups
1

I think it is doing exactly what I wanted it to do.. One capture group it has, and so it returns `1` :confused: 

From the docs :

> regex.groups
The number of capturing groups in the pattern.